product-scenarios/1-manage-public-partner-private-apis/1.4-delete-an-existing-api/1.4.1-delete-an-existing-api/src/test/java/org/wso2/am/scenario/tests/delete/existing/api/DeleteExistingAPIsTestCases.java

```diff
@@ -0,0 +1,233 @@
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.PUBLISHED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case BLOCKED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.BLOCKED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case DEPRECATED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.DEPRECATED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case RETIRED:
+                updateRequest = new APILifeCycleStateRequest(apiName, API_CREATOR_PUBLISHER_USERNAME,
+                        APILifeCycleState.RETIRED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case CREATED:
+                default:
+//                    do nothing
+        }
+    }
+
+    private void deleteAPI(String apiName) throws Exception {
+        verifyResponse(apiPublisher.deleteAPI(apiName, API_VERSION, API_CREATOR_PUBLISHER_USERNAME));
+//        verify API not available in publisher
+        HttpResponse response = apiPublisher.getAPI(apiName, API_CREATOR_PUBLISHER_USERNAME, API_VERSION);
+        log.info("API delete response code for API \'" + apiName + "\' : " + response.getResponseCode());
+        log.info("API delete response data for API \'" + apiName + "\' : " + response.getData());
+        JSONObject responseData = new JSONObject(response.getData());
+        assertTrue(responseData.getBoolean("error"), "API deletion unsuccessful for API : " + apiName);
+        assertTrue(responseData.getString("message").contains("Cannot find the requested API"),
+                "API deletion unsuccessful for API : " + apiName);
+//        verify API not available in store
+        isAPINotVisibleInStore(apiName, apiStore);
+    }
+
+    private void verifyApiStatusChange(HttpResponse apiUpdateResponse, String status) {
+        log.info("API life cycle state change response code : " + apiUpdateResponse.getResponseCode());
+        log.info("API life cycle state change response data : " + apiUpdateResponse.getData());
+        JSONArray updateStatus = new JSONObject(apiUpdateResponse.getData()).getJSONArray("lcs");
+        assertTrue(updateStatus.get(updateStatus.length() - 1).toString().contains("\"newStatus\":\"" + status + "\""),
+                "API life cycle state change failed");
+    }
+
+    private void createApplication(String applicationName) throws Exception{
```

Comment:
Isn't this test case out of the scope of this test class? As this is for deleting apis. Within this helper method you have done asserting.

Reply:
These are methods that are re-used by the scenario test cases. Assertions are added for debugging and analyzing in case the tests fail in testgrid environment.